test/functional/feature_framework_startup_failures.py

```diff
@@ -0,0 +1,155 @@
+from test_framework.test_framework import BitcoinTestFramework
+
+from hashlib import md5
+import re
+import subprocess
+import sys
+import time
+
+class FeatureFrameworkStartupFailures(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        # Don't start the node yet, as we want to measure during run_test.
+        self.add_nodes(self.num_nodes, self.extra_args)
+
+    # Launches a child test process running this same file, but instantiating
+    # a child test, and verifies that we (only) raise the expected exception.
+    def _verify_startup_failure(self, test, internal_args, expected_exception):
+        args = [
+            sys.executable,
+            __file__,
+            f"--cachedir={self.options.cachedir}",
+            f"--portseed={self.options.port_seed}",
+            f"--internal_test={test.__name__}",
+        ] + internal_args
+        if self.options.nocleanup:
+            args.append("--nocleanup")
+        weak_hash = md5(expected_exception.encode("utf-8"))
+        if self.options.tmpdir:
+            args.append(f"--tmpdir={self.options.tmpdir}/{weak_hash.hexdigest()[:16]}")
+        if self.options.randomseed is not None:
+            args.append(f"--randomseed={self.options.randomseed ^ int.from_bytes(weak_hash.digest()[:8], 'little')}")
+        if self.options.timeout_factor is not None:
+            args.append(f"--timeout-factor={self.options.timeout_factor}")
+
+        try:
+            output = subprocess.run(args,
+                                    encoding="utf-8",
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.STDOUT,
+                                    timeout=60 * self.options.timeout_factor).stdout
+        except subprocess.TimeoutExpired as e:
+            # Warning: timeouts like this halt execution of TestNode logic,
+            # meaning dangling bitcoind processes are to be expected.
+
+            # e.output is returned as bytes on Linux and str on Windows.
+            child_output = e.output.decode("utf-8") if isinstance(e.output, bytes) else e.output
+            print(f"Unexpected child timeout:\n<CHILD OUTPUT BEGIN>\n{child_output}\n<CHILD OUTPUT END>", file=sys.stderr)
+            raise
+
+        errors = []
+        if (n := output.count("Traceback")) != 1:
+            errors.append(f"Found {n}/1 tracebacks - expecting exactly one with no knock-on exceptions.")
+        if (n := len(re.findall(expected_exception, output))) != 1:
+            errors.append(f"Found {n}/1 occurrences of the specific exception: {expected_exception}")
+        if (n := len(re.findall("Test failed. Test logging available at", output))) != 1:
+            errors.append(f"Found {n}/1 test failure output messages.")
+
+        assert not errors, f"Child test didn't contain (only) expected errors:\n{chr(10).join(errors)}\n<CHILD OUTPUT BEGIN>\n{output}\n<CHILD OUTPUT END>\n"
```

Comment:
nit: '\n' seems less cryptic:
```suggestion
        assert not errors, f"Child test didn't contain (only) expected errors:\n{'\n'.join(errors)}\n<CHILD OUTPUT BEGIN>\n{output}\n<CHILD OUTPUT END>\n"
```

Reply:
This was the result of an unfortunate battle with the linter. Seems to not even have worked on older versions:

https://stackoverflow.com/questions/51775950/why-isnt-it-possible-to-use-backslashes-inside-the-braces-of-f-strings-how-can

Reply:
I haven't fully given up on this :D. It's not very important, just had to find a solution, feel free to ignore.
Since empty strings are also falsy, the 🦭 operator can come to the rescue again.
```suggestion
        assert not (msg := '\n'.join(errors)), f"Child test didn't contain (only) expected errors:\n{msg}\n<CHILD OUTPUT BEGIN>\n{output}\n<CHILD OUTPUT END>\n"
```

Reply:
Switched to use `os.linesep` which I think is more idiomatic... But I'm not entirely happy with how I ended up having to escape:
https://github.com/bitcoin/bitcoin/blob/b19439b1eee57041fc37dcf509d5ee9f248263bc/test/functional/feature_framework_startup_failures.py#L76-L79

It's a side-effect of using `repr(e)` in 28e282ef9ae94ede4aace6b97ff18c66cb72a001 which gives the escaped form.